### DISCUSSION TITLE
Limite Children Zigbee

### DISCUSSION THREAD

**Discussion by NuccioT** [indomus/forum#2044] [ZigBee, Z-Wave, Thread, Matter]
Buongiorno,
e complimenti per il vostro contributo allo sviluppo della domotica in Italia!
Ho installato una rete Zigbee con ZHA e come coordinatore ho un Eachen eWeLink ZbBridge Pro Ethernet Zigbee Gateway con Tasmota. La scelta è ricaduta su questo prodotto per via della porta ethernet. Con l'integrazione ZHA per HA Sono arrivato a 31 dispositivi collegati ma non riesco più ad aggiungerne altri. Man mano che ha aggiunto dispositivi ho sempre cercato di aggiungerli tramite quelli collegati alla rete elettrica che dovrebbero fare da router e non intasare il coordinator. Sembra invece che tutti i dispositivi si siano collegati direttamente al coordinator facendomi raggiungere il limite dei 30 dispositivi collegabili.
I device collegati sono tutti Sonoff (switch, termometri, rilevatori di movimento).
Avete idea di cosa possa fare per continuare ad aggiungere devices?
Grazie

**Comment by snakuzzo** [comment]
Da quello che leggo dovrebbe consentire il pairing di circa 120 device.
Il pairing fallo sempre con il coordinator e poi li sposti dove ti pare. Considera che non scegli tu se farli collegare o meno ai router...devi solo lasciarli lì ed aspettare. E' tutto gestito automaticamente.

**Reply by NuccioT** [reply]
Grazie per l'attenzione. Sono confuso: quando su ZHA/HA leggo "Aggiungi dispositivi tramite questo dispositivo" cosa significa? Da poco esperto penso che l'aggiunta del nuovo device avvenga tramite il dispositivo selezionato per l'aggiunta alla rete.
Se non sono io a scegliere a che dispositivo collegare i nuovi devices perché viene data questa opzione? Come faccio a spostarli dopo? Ho visto che c'è un'opzione sull'integrazione che consente di "migrare la radio": è questo che intendi? Non l'ho mai provata per timore di perdere tutto.
In effetti la mia scelta era ricaduta su quel coordinator anche in funzione del numero di devices collegabili: sono a un quarto di quello che vorrei realizzare. Stavo valutando l'acquisto di 3/4 dongle Sonoff V2 per collegarne uno al RasPI e gli altri come ripetitori ma prima di fare l'investimento vorrei essere certo. Se non ho possibilità di selezionare a che router devono connettersi anche in questo caso rischio di essere soggetto al limite?
Abusando del vostro tempo ho letto che spegnendo per 30 min il coordinator il mesh si ricostruisce ma ho timore di perdere tutto il lavoro fatto: è un tentativo che vale la pena fare?
Non so se le info fornite sono sufficienti, posso allegare uno screenshot della rete se utile.
Grazie.

**Reply by snakuzzo** [reply]
Non ti è chiaro quello che stai facendo
"Aggiungi tramite questo dispositivo" è inteso "tramite il coordinator". Quando devi aggiungere un dispositivo clicchi lì abilitando il pairing sul coordinator. A quel punto metti in pairing il device che vuoi collegare ed in automatico verrà aggiunto alla rete zigbee.
Fine.

La cosa della migrazione legata all'antenna (radio) non c'entra nulla con tutto ciò. Quello serve per migrare le impostazioni della tua rete nel caso in cui dovessi cambiare dispositivo

**Reply by NuccioT** [reply]
Grazie mille. Quindi se:
1) l'aggiunta di nuovi devices è sempre tramite coordinator
2) la configurazione della rete mesh è affidata all'integrazione/hardware

come faccio a spostare degli end devices (o anche altri router distanti dal coordinator) su routers alimentati da rete per non appesantire il coordinator?
Perché tutti i miei devices (router ed end devices) sono finiti sul coordinator? 
La migrazione dovrei utilizzarla se decidessi di installare il dongle Sonoff?
Grazie ancora

**Reply by snakuzzo** [reply]
Ripeto...la rete è gestita in autonomia dal coordinator. Tu non hai potere decisionale

**Reply by snakuzzo** [reply]
È il coordinator a gestire il tutto

**Reply by NuccioT** [reply]
ok questo è assodato. Nella tua prima risposta però leggo: "Il pairing fallo sempre con il coordinator e poi li sposti dove ti pare"; è questo che non mi è chiaro, come li sposto?
Se è il coordinator a gestire è lecito pensare che alcuni lo facciano meglio di altri oppure la distribuzione dei devices segue uno standard (Zigbee Alliance)?

**Reply by snakuzzo** [reply]
Intendo dire che mettiti col device vicino al coordinator e fai il pairing. Quando l'hai fatto mettiil i device dove vuoi che siano piazzati.

I coordinator possono avere firmware diversi (a seconda anche dell'hardware) e si basano su specifici SDK. Lo standard non è sempre rispettato alla lettera da chi costruisce e distribuisce gli end device.

**Reply by NuccioT** [reply]
Ok, grazie mille, provo e ti faccio sapere.

**Reply by NuccioT** [reply]
Anche se da: https://www.home-assistant.io/integrations/zha#best-practices-to-avoid-pairingconnection-difficulties

BEST PRACTICES TO AVOID PAIRING/CONNECTION DIFFICULTIES
Verify that you try to follow recommended best practices to avoid pairing and/or connection issues:

If possible try to pair your Zigbee devices in their intended final location, (and **not pair it next to the Zigbee coordinator and then need to move it after**).
Pairing a Zigbee device next to the Zigbee coordinator and then moving it later can result in dropped/lost connections or other issues.

**Reply by snakuzzo** [reply]
Bah...posso dirti che ho sempre fatto il pairing in prossimità del coordinator e poi ho spostato i device ... e non ho mai avuto problemi

**Reply by Scialla** [reply]
Aggiungo che, anche se facessi il pairing su un router (è possibile ma a volte difficoltoso), la rete poi si autogestisce collegando l'end device dove vuole (dando priorità sempre al coordinator).
Quindi meglio il pairing sul coordinator da vicino, poi lo sposti dove vuoi e la rete si adatta

**Reply by NuccioT** [reply]
Grazie, stasera provo e vi saprò dire.

**Reply by NuccioT** [reply]
OK, abbinato con coordinator e spostato: funziona. L'unico valore sballato è la batteria. Ho acquistato e installato questo sensore insieme ad altri e il livello della batteria dovrebbe essere simile. Invece mi dà 5% contro il 75%-80% degli altri sensori. Quando funzionava i valori erano simili.

**Reply by snakuzzo** [reply]
il livello batteria, nonostante tutti gli aggiustamenti fatti nel tempo, è sempre un valore calcolato in base ai V... non è un valore attendibilissimo
